File: fm-domain/src/main/java/io/zeitwert/fm/portfolio/model/ObjPortfolioRepository.java
package io.zeitwert.fm.portfolio.model;

import io.dddrive.core.obj.model.ObjRepository;
import io.zeitwert.fm.account.model.ObjAccountRepository;
import io.zeitwert.fm.building.model.ObjBuildingRepository;
import io.zeitwert.fm.task.model.DocTaskRepository;

public interface ObjPortfolioRepository extends ObjRepository<ObjPortfolio> {

	ObjAccountRepository getAccountRepository();

	ObjBuildingRepository getBuildingRepository();

	DocTaskRepository getTaskRepository();

}

File: fm-domain/src/main/java/io/zeitwert/fm/portfolio/model/ObjPortfolioRepository.kt
package io.zeitwert.fm.portfolio.model

import io.zeitwert.fm.account.model.ObjAccountRepository
import io.zeitwert.fm.building.model.ObjBuildingRepository
import io.zeitwert.fm.obj.model.FMObjRepository
import io.zeitwert.fm.task.model.DocTaskRepository

interface ObjPortfolioRepository : FMObjRepository<ObjPortfolio> {

	val accountRepository: ObjAccountRepository

	val buildingRepository: ObjBuildingRepository

	val taskRepository: DocTaskRepository

}
